appcues/src/main/java/com/appcues/statemachine/StateMachineFactory.kt

```diff
@@ -0,0 +1,16 @@
+package com.appcues.statemachine
+
+import com.appcues.AppcuesCoroutineScope
+import com.appcues.action.ActionProcessor
+import com.appcues.data.model.RenderContext
+import com.appcues.statemachine.StateMachine.StateMachineListener
+
+internal class StateMachineFactory(
```

Comment:
a Factory of StateMachines, the create method receives the renderContext and a StateMachineListener (this will be refactored later, since we will be moving away from using this StateMachineListener and revert back to StateFlow